algorithm-simulator/src/main/scala/style95/generator/Transition.scala

```diff
@@ -0,0 +1,38 @@
+package style95.generator
+
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.math.{log, random}
+
+case class Transition(startTps: Int, endTps: Int) {
+  // Use `in` method to make the DSL of scenario builder consistent.
+  def in(duration: FiniteDuration): CompleteTransition =
+    new CompleteTransition(startTps, endTps, duration)
+}
+
+// Users do not need to construct this class manually.
+class CompleteTransition private[generator](startTps: Int,
+                                            endTps: Int,
+                                            val duration: FiniteDuration)
+    extends IntervalBehavior {
+  class Behavior extends BehaviorDescriptor {
+    class Gen extends TimingGenerator {
+      def next(elapsed: FiniteDuration): FiniteDuration = {
+        require(startTps >= 0 && endTps >= 0, "tps should be positive")
+
+        val tps = lerp(startTps, endTps, elapsed / duration)
+        // avoid log(0)
+        // avoid zero division
+        // please refer to https://preshing.com/20111007/how-to-generate-random-timings-for-a-poisson-process/
+        (-log(1.0 - random()) / (tps + 1e-6)) seconds
```

Comment:
why is `1e-6` required here?

Reply:
It's used for preventing from dividing by zero.
In the testing, I found some inopportune timing that made the lerped `tps` equal or very close to zero. For example, it can happen at the very first moment in a transition from 0 TPS to 10 TPS. 
Adding a small constant to the divisor is a common trick to prevent numeric unstability.

Reply:
ok then doesn't the same thing happen when we maintain a small TPS here?
https://github.com/style95/queue-simulator/pull/3/files#diff-1e65bacabd108a8a9c0daca35dc36d05R13

Reply:
Well, it is an inconsistency. Do you think that we should change the argument `tps` of `Maintain` from `Int` to `Double`? If `tps` remains an integer, we don't need to add a small constant to the divisor. And if `tps` should be a double, the small constant is required.

Reply:
I think it is reasonable to use `Int` for TPS as per its definition, "Transactions Per Second".
How about changing the prerequisite to require bigger than 1 start and end TPS?
And if I understand correctly, if `startTps == endTps`, should it work like `Maintain`?

Reply:
Agreed. TPS should be an integer by its name.
`Transition` works exactly as `Maintain` if `startTps == endTps`.

Reply:
Then we may need the logic for it here? https://github.com/style95/queue-simulator/pull/3/files/97f31c6c0d69c882f1ed3c6f1ca640a99080e8ea#diff-7d55a7ce4575a8b88acf5b890b6b6eecR31

Reply:
Lerp should work fine when `startTps == endTps`. `t * (end - start) + start` will degrade to `start` when `startTps == endTps`, which essentially is an `Maintain(start).
`

Reply:
Got it.